clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/remote/RemoteCacheContainerConfigurationBuilder.java

```diff
@@ -0,0 +1,187 @@
+        this.maxRetries = Attribute.MAX_RETRIES.resolveModelAttribute(context, model).asInt();
+        this.protocolVersion = Attribute.PROTOCOL_VERSION.resolveModelAttribute(context, model).asString();
+        this.socketTimeout = Attribute.SOCKET_TIMEOUT.resolveModelAttribute(context, model).asInt();
+        this.tcpNoDelay = Attribute.TCP_NO_DELAY.resolveModelAttribute(context, model).asBoolean();
+        this.tcpKeepAlive = Attribute.TCP_KEEP_ALIVE.resolveModelAttribute(context, model).asBoolean();
+        this.valueSizeEstimate = Attribute.VALUE_SIZE_ESTIMATE.resolveModelAttribute(context, model).asInt();
+
+        this.clusters.clear();
+        if (model.hasDefined(RemoteClusterResourceDefinition.WILDCARD_PATH.getKey())) {
+            for (Property property : model.get(RemoteClusterResourceDefinition.WILDCARD_PATH.getKey()).asPropertyList()) {
+                String clusterName = property.getName();
+                List<InjectedValueDependency<OutboundSocketBinding>> valueDependencies = new ArrayList<>();
+                for (String binding : StringListAttributeDefinition.unwrapValue(context, RemoteClusterResourceDefinition.Attribute.SOCKET_BINDINGS.resolveModelAttribute(context, property.getValue()))) {
+                    InjectedValueDependency<OutboundSocketBinding> outboundSocketBindingInjectedValueDependency = new InjectedValueDependency<>(CommonUnaryRequirement.OUTBOUND_SOCKET_BINDING.getServiceName(context, binding), OutboundSocketBinding.class);
+                    valueDependencies.add(outboundSocketBindingInjectedValueDependency);
+                }
+                this.clusters.put(clusterName, valueDependencies);
+            }
+        }
+
+        return this;
+    }
+
+    @Override
+    public ServiceBuilder<Configuration> build(ServiceTarget target) {
+        ServiceBuilder<Configuration> builder = target.addService(this.getServiceName(), new ValueService<>(this))
+                .addDependency(RemoteCacheContainerComponent.CONNECTION_POOL.getServiceName(this.address), ConnectionPoolConfiguration.class, this.connectionPool)
+                .addDependency(RemoteCacheContainerComponent.NEAR_CACHE.getServiceName(address), NearCacheConfiguration.class, this.nearCache)
+                .setInitialMode(ServiceController.Mode.ON_DEMAND)
+                ;
+        for (ValueDependency<ExecutorFactoryConfiguration> dependency : this.threadPools.values()) {
+            dependency.register(builder);
+        }
+        for (List<InjectedValueDependency<OutboundSocketBinding>> dependencies : this.clusters.values()) {
+            for (InjectedValueDependency<OutboundSocketBinding> dependency : dependencies) {
+                dependency.register(builder);
+            }
+        }
+
+        return builder;
+    }
+
+    @Override
+    public Configuration getValue() {
+        ConfigurationBuilder builder = new ConfigurationBuilder()
```

Comment:
Additionally, we need to configure the Marshaller.  Similiar to what is done here:
https://github.com/wildfly-clustering/wildfly-clustering-web-hotrod/blob/master/src/main/java/org/wildfly/clustering/web/hotrod/HotRodMarshaller.java
Except that it should use a ModularClassResolver.

Reply:
Done; added new commit to make review easier.